Ginkgo CI: re-running tests locally after VM is already provisioned sporadically fails

After running the tests and having the runtime test VM boot, when I re-run the tests, the provisioning step fails:

```
Ians-MBP:test ianvernon$ ginkgo --focus="RuntimeMonitorTest" -v -noColor
time="2017-11-30T11:44:10-08:00" level=info msg="environment variable \"K8S_VERSION\" was not set; setting to default value \"1.7\""
Running Suite: runtime
======================
Random Seed: 1512071046
Will run 6 of 47 specs

time="2017-11-30T11:44:10-08:00" level=info msg="Vagrant: running command \"vagrant status runtime --machine-readable\""
time="2017-11-30T11:44:13-08:00" level=info msg="build is not running on Jenkins; environment variable \"JENKINS_HOME\" is not set"
time="2017-11-30T11:44:13-08:00" level=info msg="build is not running on Jenkins; environment variable \"NODE_NAME\" is not set"
time="2017-11-30T11:44:13-08:00" level=info msg="Vagrant:Create: running 'vagrant provision runtime'"
time="2017-11-30T11:44:13-08:00" level=info msg="Vagrant: running command \"vagrant provision runtime\""
time="2017-11-30T11:44:15-08:00" level=info msg="stdout: ==> runtime: Running provisioner: shell..."
time="2017-11-30T11:44:16-08:00" level=info msg="stdout:     runtime: Running: inline script"
time="2017-11-30T11:44:16-08:00" level=info msg="stdout: ==> runtime: Running provisioner: file..."
time="2017-11-30T11:44:17-08:00" level=info msg="stdout: ==> runtime: Running provisioner: shell..."
time="2017-11-30T11:44:17-08:00" level=info msg="stdout:     runtime: Running: /var/folders/5f/63bp05l16w33ts7rsybrn_w00000gn/T/vagrant-shell20171130-92808-19dnpto.sh"
time="2017-11-30T11:44:17-08:00" level=info msg="stdout: ==> runtime: The user `vagrant' is already a member of `docker'."
time="2017-11-30T11:44:18-08:00" level=info msg="stdout: ==> runtime: # cd /go/src/github.com/jteeuwen/go-bindata; git pull --ff-only"
time="2017-11-30T11:44:18-08:00" level=info msg="stdout: ==> runtime: fatal: unable to access 'https://github.com/jteeuwen/go-bindata/': Could not resolve host: github.com"
time="2017-11-30T11:44:18-08:00" level=info msg="stdout: ==> runtime: package github.com/jteeuwen/go-bindata: exit status 1"
time="2017-11-30T11:44:18-08:00" level=info msg="stderr: The SSH command responded with a non-zero exit status. Vagrant"
time="2017-11-30T11:44:18-08:00" level=info msg="stderr: assumes that this means the command failed. The output for this command"
time="2017-11-30T11:44:18-08:00" level=info msg="stderr: should be in the log above. Please read the output to determine what"
time="2017-11-30T11:44:18-08:00" level=info msg="stderr: went wrong."
Failure [7.554 seconds]
[BeforeSuite] BeforeSuite 
/Users/ianvernon/go/src/github.com/cilium/cilium/test/test_suite_test.go:111

  error starting VM "runtime": exit status 1

  /Users/ianvernon/go/src/github.com/cilium/cilium/test/test_suite_test.go:128
------------------------------
time="2017-11-30T11:44:18-08:00" level=info msg="build is not running on Jenkins; environment variable \"JENKINS_HOME\" is not set"
time="2017-11-30T11:44:18-08:00" level=info msg="build is not running on Jenkins; environment variable \"NODE_NAME\" is not set"
time="2017-11-30T11:44:18-08:00" level=info msg="AfterSuite: not running on Jenkins; leaving VMs running for debugging"

Ran 6 of 0 Specs in 7.554 seconds
FAIL! -- 0 Passed | 6 Failed | 0 Pending | 0 Skipped --- FAIL: TestTest (7.56s)
FAIL

Ginkgo ran 1 suite in 11.996047233s
Test Suite Failed
```

--- a/test/provision/dns.sh
+++ b/test/provision/dns.sh
@@ -1,0 +1,10 @@
+#!/bin/bash
+
+# This script update the dns servers to use google ones.
+set -e
+
+cat <<EOF > /etc/resolv.conf
+nameserver 8.8.8.8
+nameserver 8.8.4.4
+EOF
+

--- a/test/provision/k8s_install.sh
+++ b/test/provision/k8s_install.sh
@@ -4,6 +4,7 @@ set -e
 HOST=$(hostname)
 TOKEN="258062.5d84c017c9b2796c"
 CILIUM_CONFIG_DIR="/opt/cilium"
+PROVISIONSRC="/tmp/provision/"
 ETCD_VERSION="v3.1.0"
 NODE=$1
 IP=$2
@@ -16,6 +17,8 @@ if [[ -f  "/etc/provision_finished" ]]; then
     /tmp/provision/compile.sh
     exit 0
 fi
+
+$PROVISIONSRC/dns.sh
 
 cat <<EOF > /etc/hosts
 127.0.0.1       localhost
@@ -65,7 +68,7 @@ if [[ "${HOST}" == "k8s1" ]]; then
     kubectl taint nodes --all node-role.kubernetes.io/master-
 
     sudo systemctl start etcd
-    /tmp/provision/compile.sh
+    $PROVISIONSRC/compile.sh
 else
     kubeadm join --token=$TOKEN 192.168.36.11:6443
     cp /etc/kubernetes/kubelet.conf ${CILIUM_CONFIG_DIR}/kubeconfig

--- a/test/provision/runtime_install.sh
+++ b/test/provision/runtime_install.sh
@@ -2,11 +2,12 @@
 set -e
 
 HOST=$(hostname)
+PROVISIONSRC="/tmp/provision/"
 
+$PROVISIONSRC/dns.sh
 sudo adduser vagrant docker
 
 export GOPATH=/go/
 go get -u github.com/jteeuwen/go-bindata/...
 ln -sf /go/bin/* /usr/local/bin/
-/tmp/provision/compile.sh
-
+$PROVISIONSRC/compile.sh

